test_naucse/test_validation.py

```diff
@@ -1,100 +1,116 @@
-import pytest
+from textwrap import dedent
 
-import naucse.validation
+import pytest
 
+import naucse.sanitize
+from naucse.sanitize import sanitize_html
 
-def test_allowed_elements():
-    allowed_elements = naucse.validation.AllowedElementsParser()
+def assert_changed(input_html, expected):
+    input_html = dedent(input_html).strip()
+    expected = dedent(expected).strip()
+    output_html = sanitize_html(input_html)
+    assert output_html == expected
 
-    allowed_elements.reset_and_feed(
-        "<div><strong><u><a>Test</a></u></div>"
-    )
 
-    with pytest.raises(naucse.validation.DisallowedElement):
-        allowed_elements.reset_and_feed(
-            "<div><script>alert('XSS')</script></div>"
-        )
+def assert_unchanged(input_html):
+    assert_changed(input_html, input_html)
 
 
-def test_allow_attributes():
-    allowed_elements = naucse.validation.AllowedElementsParser()
-
-    allowed_elements.reset_and_feed(
-        "<div class='test'><span style='color: red'><a href='/courses/'>Text</a></span></div>"
+def test_allowed_elements():
+    assert_changed(
+        "<div><strong><u><a>Test</a></u></div>",
+        "<div><strong><u><a>Test</a></u></strong></div>",
```

Comment:
where is this change coming form? I might have just missed it, but I don't get it. si it a side effect of parsing HTML with lxml and dumping it back to string?

Reply:
Yes, this comes from the lxml round-trip.

Reply:
consider adding a comment about it, as it is not so obvious.